pycbc/inject/inject.py

```diff
@@ -936,7 +965,8 @@ def slice_and_taper(inj, ts):
         return ts
 
     def apply(self, strain, detector_name, distance_scale=1,
-              injection_sample_rate=None, inj_filter_rejector=None):
+              injection_sample_rate=None, inj_filter_rejector=None,
+              generate_injections=True):
         if inj_filter_rejector is not None:
             raise NotImplementedError("IncoherentFromFile injections do not "
```

Comment:
The 'generate_injections' part seems like a bit of hack. Is there a reason to do this? It seems like the logic is that some setup is needed for the injection filter rejection, but it seems to add a lot of fiddly bits. It seems very unintuitive what is happening when the 'generate_injections' option is set to false. I would prefer this option be removed. 

For injection frames, it would seem you should simply not call this method. And for injection filter rejector you essentially use all the parts of the apply method, so the if statements just complicate what is happening. I'm not sure any computation is really saved since the full injection has to be made in this case. 

Overall, the logic seems like it would be more clear if not hidden in this function, and it would take up less space.